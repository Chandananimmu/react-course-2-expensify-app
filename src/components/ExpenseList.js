import React from "react";
import { connect } from "react-redux";
import ExpenseListItem from "./ExpenseListItem";
import getVisibleExpenses from "../selectors/expenses";
export const ExpenseList = props => (
  <div className="content-container">
    <div className="list-header">
      <div className="show-for-mobile">Expenses</div>
      <div className="show-for-desktop">Expense</div>
      <div className="show-for-desktop">Amount</div>
  
</div>
<div className="list-body">
    {props.expenses.length === 0 ? (
      <div>
         <span className="list-item list-item--message">No expenses</span>
      </div>
     
    ) : (
      props.expenses.map(expense => {
        return <ExpenseListItem key={expense.id} {...expense} />;
      })
    )}
    </div>
  </div>
);
const mapToStateProps = state => {
  return {
    expenses: getVisibleExpenses(state.expenses, state.filters)
  };
};

//const ConnnectedList = connect(mapToStateProps)(ExpenseList);
export default connect(mapToStateProps)(ExpenseList);
