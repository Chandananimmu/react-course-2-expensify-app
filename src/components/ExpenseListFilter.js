import React from "react";
import { connect } from "react-redux";
import { DateRangePicker } from "react-dates";
import {
  setFilterText,
  sortByDate,
  sortByAmount,
  setStartDate,
  setEndDate
} from "../actions/filters";

export class ExpenseListFilter extends React.Component {
  state = {
    calenderFocused: null
  };
  onDatesChange = ({ startDate, endDate }) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
  };
  onFocusChange = calenderFocused => {
 
    this.setState(() => ({
      calenderFocused
    }));
  };
  onTextChange = e => {
    this.props.setFilterText({ text: e.target.value });
  };
  onSortChange = e => {
    e.target.value === "date"
      ? this.props.sortByDate()
      : this.props.sortByAmount();
  };
  render() {
    return (
      <div>
        <input
          type="text"
          value={this.props.filters.text}
          // onChange={e => {
          //   this.props.dispatch(setFilterText({ text: e.target.value }));
          // }}
          onChange={this.onTextChange}
        />
        <select
          value={this.props.filters.sortBy}
          // onChange={e => {
          //   this.props.dispatch(
          //     e.target.value === "date" ? sortByDate() : sortByAmount()
          //   );
          // }}
          onChange={this.onSortChange}
        >
          <option value="date">Date</option>
          <option value="amount">Amount</option>
        </select>
        <DateRangePicker 
          startDate={this.props.filters.startDate}
          endDate={this.props.filters.endDate}
          onDatesChange={this.onDatesChange}
          focusedInput={this.state.calenderFocused}
          onFocusChange={this.onFocusChange}
          numberOfMonths={1}
          isOutsideRange={() => false}
          showClearDates={true}
          
        />
      </div>
    );
  }
}
//const ExpenseListFilter = props =>
const mapStateToProps = state => {
  return {
    filters: state.filters
  };
};
const mapDispatchToProps = dispatch => ({
  setStartDate: startDate => {
    dispatch(setStartDate(startDate));
  },
  setEndDate: endDate => {
    dispatch(setEndDate(endDate));
  },
  setFilterText: text => {
    dispatch(setFilterText(text));
  },
  sortByDate: () => {
    dispatch(sortByDate());
  },
  sortByAmount: () => {
    dispatch(sortByAmount());
  }
});
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ExpenseListFilter);
