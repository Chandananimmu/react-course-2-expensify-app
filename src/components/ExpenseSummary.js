import React from "react";
import { connect } from "react-redux";
import getVisibleExpenses from "../selectors/expenses";
import getExpensesTotal from "../selectors/expenses-total";
import numeral from "numeral";
export const ExpenseSummary = props => (
  <div>
    <p>
      Viewing {props.expenses.length} expenses totalling{" "}
      {numeral(getExpensesTotal(props.expenses) / 100).format("$0,0.00")}
    </p>
  </div>
);
const mapToStateProps = state => {
  return {
    expenses: getVisibleExpenses(state.expenses, state.filters),
  };
};
export default connect(mapToStateProps)(ExpenseSummary);
