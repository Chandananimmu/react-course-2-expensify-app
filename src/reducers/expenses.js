const expensesReducerDefaultState = [];
export default (state = expensesReducerDefaultState, action) => {
  switch (action.type) {
    case "ADD_EXPENSE":
      return [...state, action.expense];
    case "REMOVE_EXPENSE":
      const idd = action.id;
      return state.filter(({ id }) => id !== idd); //idd->the one wich v need to remove
    case "EDIT_EXPENSE":
      return state.map(expense => {
        if (expense.id === action.id) {
          return {
            ...expense,
            ...action.updates
          };
        } else return expense;
      });

    default:
      return state;
  }
};