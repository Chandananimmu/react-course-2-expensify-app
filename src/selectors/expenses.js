import moment from "moment";
const getVisibleExpenses = (
  expenses, {
    text,
    sortedBy,
    startDate,
    endDate
  }
) => {
  return expenses
    .filter(expense => {
      const createdAtMoment = moment(expense.createdAt)
      const startDateMatch =
        startDate ? startDate.isSameOrBefore(createdAtMoment, 'day') : true;
      const EndDateMatch =
        endDate ? endDate.isSameOrAfter(createdAtMoment, 'day') : true;
      //const startDateMatch=true;
      //const EndDateMatch=true
      //const t1=expense.description.toLowerCase();
      //const t2=text.toLowerCase();
      const textMatch = expense.description
        .toLowerCase()
        .includes(text.toLowerCase());

      return startDateMatch && EndDateMatch && textMatch;
    }).sort((a, b) => {
      if (sortedBy === 'date') {
        return a.createdAt < b.createdAt ? 1 : -1;
      } else
      if (sortedBy === 'amount') {
        return a.amount < b.amount ? 1 : -1;
      }
    });

};
export default getVisibleExpenses;