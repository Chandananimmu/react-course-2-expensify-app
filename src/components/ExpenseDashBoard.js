import React from "react";
import ExpenseList from "./ExpenseList";
import ExpenseListFilter from "./ExpenseListFilter";
import ExpenseSummary from "./ExpenseSummary";
const ExpenseDashBoard = () => (
  <div>
    <ExpenseSummary />
    <ExpenseList />
    <ExpenseListFilter />
  </div>
);
export default ExpenseDashBoard;
