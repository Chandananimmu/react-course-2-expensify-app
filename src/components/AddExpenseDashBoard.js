import React from "react";
import { connect } from "react-redux";
import ExpenseForm from "./ExpenseForm";
import { addExpense } from "../actions/expenses";
export class AddExpenseDashBoard extends React.Component {
  onSubmit = (expense) => {
    this.props.addExpense(expense);
    this.props.history.push('/');
  };
  render() {
    return (
      <div>
        <p>This msg is from add dashboard </p>
        <ExpenseForm onSubmit={this.onSubmit} />
      </div>
    );
  }
}
// const AddExpenseDashBoard = props => (
//   <div>
//     <p>This msg is from add dashboard </p>
//     <ExpenseForm
//       onSubmit={expense => {
//         //  console.log(expense);
//         props.onsubmit(expense);
//         props.history.push("/");
//       }}
//     />
//   </div>
// );
const mapDispatchToProps = (dispatch) => ({
  addExpense: (expense) => dispatch(addExpense(expense))
});
export default connect( undefined, mapDispatchToProps) (AddExpenseDashBoard);
 


