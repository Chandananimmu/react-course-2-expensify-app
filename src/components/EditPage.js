import React from "react";
import ExpenseForm from "./ExpenseForm";
import { connect } from "react-redux";
import { editExpense } from "../actions/expenses";
import { removeExpense } from "../actions/expenses";
export class EditPage extends React.Component {
  onSubmit =( expense) => {
    this.props.editExpense(this.props.expense.id, expense);
    this.props.history.push("/");
  };
  onClick = () => {
    this.props.removeExpense({ id: this.props.expense.id });
    this.props.history.push("/");
  };
  render() {
    return (
      <div>
        <ExpenseForm expense={this.props.expense} onSubmit={this.onSubmit} />
        <button onClick={this.onClick}>Remove</button>
      </div>
    );
  }
}
// const EditPage = props => (
//   <div>
//     <ExpenseForm
//     expense={props.expense}
//       onSubmit={(expense) => {
//        const id=props.expense.id
//         props.dispatch(editExpense(id,expense));
//         props.history.push("/")
//        // console.log("updated", expense);
//       }}
//     />
//      <button
//       onClick={() => {
//         const id=props.expense.id
//         props.dispatch(removeExpense({id}));
//         props.history.push("/")
//       }}
//     >
//       Remove
//     </button>
//   </div>
// );

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find(
      expense => expense.id === props.match.params.id
    )
  };
};
const mapDispatchToProps = (dispatch, props) => ({
  editExpense: (id, expense) => dispatch(editExpense(id, expense)),
  removeExpense: id => dispatch(removeExpense(id))
});
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(EditPage);
