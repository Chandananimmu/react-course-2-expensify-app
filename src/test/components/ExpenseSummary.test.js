import { shallow } from "enzyme";
import React from "react";
import expenses from "../fixtures/expenses";
import { ExpenseSummary} from  "../../components/ExpenseSummary";
test('shud render no. of expenses',()=>{
  const wrapper=shallow(<ExpenseSummary expenses={[expenses[1]]} />);
  expect(wrapper).toMatchSnapshot();

});
test('shud render multiple of expenses',()=>{
  const wrapper=shallow(<ExpenseSummary expenses={expenses} />);
  expect(wrapper).toMatchSnapshot();

});
