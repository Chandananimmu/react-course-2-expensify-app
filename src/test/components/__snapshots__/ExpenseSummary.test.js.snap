// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shud render multiple of expenses 1`] = `
<div>
  <p>
    Viewing 
    3
     expenses totalling
     
    $198.75
  </p>
</div>
`;

exports[`shud render no. of expenses 1`] = `
<div>
  <p>
    Viewing 
    1
     expenses totalling
     
    $195.00
  </p>
</div>
`;
